src/FsCheck/Runner.fs

```diff
@@ -120,6 +122,42 @@ module Runner =
                     yield EndShrink result
               yield! test newSize resize rnd1 gen
         }
+        
+#if PCL // sorry windows phone no parallelism for you
+    let private parallelTest limit = test
+#else
+    let internal parallelTest limit initSize resize rnd0 gen =
+        (initSize, rnd0)
+        |> Seq.unfold (fun (initSize, rnd) ->
+            let rnd1, rnd2 = Random.split rnd
```

Comment:
Actually I'm crafting next version of parallel runner (third one) with algorithm behind like you described, running on bare threads with no contention on producer seq. Degree of parallelism currently supported in config (have not pushed yet, code is dirty right now :rage1:) but seems it has almost no effect on plinq. Lot of performance ruined by requirement to preserve ordering among produced results though plinq docs says there will be little degradation. My algorithm isn't affected by this since it preallocates result array upfront and each thread has it own index per iteration.